src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 4;
+    private static final int MAX_NUMBER = 20;
+    private static final ColorSupplier RANDOM_COLOR = new ColorSupplier();
+    private static final Random RANDOM_NUMBER = new Random();
+
+    public Figure getRandomFigure() {
+        Figure randomFigure;
+
+        switch (getRandomNumberOfFigure()) {
+            case 0:
+                randomFigure = new Circle(RANDOM_COLOR.getRandomColor(), getRandomParameter());
+                break;
+            case 1:
+                randomFigure = new IsoscelesTrapezoid(RANDOM_COLOR.getRandomColor(),
+                        getRandomParameter(), getRandomParameter(), getRandomParameter());
+                break;
+            case 2:
+                randomFigure = new Rectangle(RANDOM_COLOR.getRandomColor(),
+                        getRandomParameter(), getRandomParameter());
+                break;
+            case 3:
+                randomFigure = new RightTriangle(RANDOM_COLOR.getRandomColor(),
+                        getRandomParameter(), getRandomParameter());
+                break;
+            default:
+                randomFigure = new Square(RANDOM_COLOR.getRandomColor(), getRandomParameter());
+        }
```

Comment:
Lets create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability. (Remember private methods should follow after public as helper ones)

Reply:
what about this comment?